### PR TITLE
[WIP] dart-sass compatibility tweaks

### PR DESCRIPTION
**Pull request by atesgoral**
### WHY are these changes introduced?

Sass is not going away any time soon (~6 mo). It might be still worth updating to dart-sass.

### WHAT is this pull request doing?

Taking a stab at updating to dart-sass and fixing compatibility issues that pop up.

Ran automigration for division operators (to convert them to multiplication): `npx sass-migrator division src/**/*.scss`

The `"compact"` value for `outputStyle` is [not supported by dart-sass](https://sass-lang.com/documentation/js-api#outputstyle). `"expanded"` is the default. So, just removed the option.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

**Review by BPScott — COMMENTED** [review]
Yay for preparing to move to dart-sass. However there are compatibility issues with putting dart-sass only code into the public API.

We'll either need to come up with some work arounds to support usage in both node-sass and dart-sass, or remove the problematic parts of the public API before moving to dart-sass

File: src/styles/foundation/_typography.scss [hunk 1]
```diff
@@ -14,7 +14,7 @@ $font-family-data: (
   Consolas,
   Liberation Mono,
   Menlo,
-  monospace !default},
+  monospace},
```
[the remark this answers is not in the document]
Comment (atesgoral) [reply src/styles/foundation/_typography.scss:17]: #4308

File: config/rollup/plugin-styles.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ export function styles({
 
   const filter = createFilter(include, exclude);
 
-  const renderSass = promisify(nodeSass.render);
+  const renderSass = promisify(sass.render);
```
Comment (nickpresta) [inline config/rollup/plugin-styles.js:25]: We will want to use `sass.renderSync`, which is apparently the [faster of the two methods](https://github.com/sass/dart-sass#javascript-api). I did a similar thing [in here](https://github.com/Shopify/loom/pull/295)

**Comment by atesgoral** [comment]
@nickpresta Feel free to take this branch over as I had abandoned this effort a while ago...

**Comment by BPScott** [comment]
Gonna close this. It'll be easier to reopen in the future. Lots of these changes have superseded. See #4725 for thoughts on deprecation warnings.

**Pull request closed without merging**